cli/netcheck.go

```diff
@@ -0,0 +1,58 @@
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/cli/clibase"
+	"github.com/coder/coder/coderd/healthcheck"
+	"github.com/coder/coder/codersdk"
+)
+
+func (r *RootCmd) netcheck() *clibase.Cmd {
+	client := new(codersdk.Client)
+
+	cmd := &clibase.Cmd{
+		Use:    "netcheck",
+		Short:  "Print network debug information",
+		Hidden: true,
+		Middleware: clibase.Chain(
+			r.InitClient(client),
+		),
+		Handler: func(inv *clibase.Invocation) error {
+			ctx, cancel := context.WithTimeout(inv.Context(), 10*time.Second)
+			defer cancel()
+
+			connInfo, err := client.WorkspaceAgentConnectionInfo(ctx)
+			if err != nil {
+				return err
+			}
+
+			var report healthcheck.DERPReport
+			report.Run(ctx, &healthcheck.DERPReportOptions{
```

Comment:
Should probably log that it's gathering a report and may take up to e.g. 1 minute.